client/commonplugins/commonplugins.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package commonplugins
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/go-version"
+)
+
+var (
+	ErrPluginNotExists     error = errors.New("plugin not found")
+	ErrPluginNotExecutable error = errors.New("plugin not executable")
+)
+
+type CommonPlugin interface {
+	Fingerprint(ctx context.Context) (*PluginFingerprint, error)
+}
+
+// CommonPlugins are expected to respond to 'fingerprint' calls with json that
+// unmarshals to this struct.
+type PluginFingerprint struct {
+	Version *version.Version `json:"version"`
+	Type    *string          `json:"type"`
+}
+
+// runPlugin is a helper for executing the provided Cmd and capturing stdout/stderr.
+// This helper implements both the soft and hard timeouts defined by the common
+// plugins interface.
+func runPlugin(ctx context.Context, cmd *exec.Cmd, cmdTimeout, killTimeout time.Duration) (stdout, stderr []byte, err error) {
+	var outBuf, errBuf bytes.Buffer
+	cmd.Stdout, cmd.Stderr = &outBuf, &errBuf
+
+	// Start the command
+	if err = cmd.Start(); err != nil {
+		return
+	}
+
+	done := make(chan error, 1)
+	go func() {
+		done <- cmd.Wait()
+	}()
+
+	plugCtx, cancel := context.WithTimeout(ctx, cmdTimeout)
+	defer cancel()
+
+	select {
+	case <-plugCtx.Done():
+		err = cmd.Process.Signal(syscall.SIGTERM)
```

Comment:
```suggestion
		_ = cmd.Process.Signal(syscall.SIGTERM)
```

I don't see a way this `err` value ever gets used, so might as well make that clear with `_`.

Seems like the most likely case is that there's a race between `plugCtx.Done()` and the process exiting "naturally" where `plugCtx.Done()` wins, but then the process exits before the signal is sent. Your implementation handles this case elegantly because `case err = <-done` will notice the process exited, and in the happy case: set `err = nil` because the plugin completed successfully.

If the signal method returns an error for any other reason... yikes... I think the killTimer path is fine for handling whatever that scenario would be. Can't imagine what else there is to be done.